servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpProtocolConfig.java

```diff
@@ -36,4 +38,13 @@
      * @return {@link HttpHeadersFactory} to be used for creating {@link HttpHeaders} when decoding HTTP messages
      */
     HttpHeadersFactory headersFactory();
+
+    /**
+     * A collection of all {@link ContentCoding}s the endpoint supports.
+     * The list will be advertised as part of the Accept-Encoding header.
+     *
+     * @return The list of supported {@link ContentCoding}s for this endpoint.
+     * @see <a href="https://tools.ietf.org/html/rfc7231#page-41">Accept-Encodings</a>
+     */
+    Set<ContentCoding> supportedEncodings();
```

Comment:
I believe this is doable as is now. If you provide NO supported-encodings for the server, then any encoded request will fail, and you can manually enable a response encoding using the `HttpMetadata.encoding()`

Reply:
Agreed, this is a way to work around the issue, but it looks like a most-common use-case for the server-side to have a compression for responses only.
It looks like if we move this configuration to the client/server builder, we will have better control. For example, server's builder may have 2 overloads: one to support compression in both ways, another overload to support it only for responses.

Having it on the builder will also help:
1. to avoid duplication. Hard to imagine that users need compression for h1, but don't need it for h2. Therefore, makes more sense to keep them applied for all supported transport protocols.
2. to consolidate logic in the filter instead of expanding responsibility of `NettyHttpServer` and client connections.

Reply:
Btw, we should also account for the proxy use-case: a gateway proxy service that resends the response from downstream backend as-is, when the response is already compressed.